doc/classes/Viewport.xml

```diff
@@ -317,11 +323,26 @@
 			Do not perform any antialiasing in the full screen post-process.
 		</constant>
 		<constant name="SCREEN_SPACE_AA_FXAA" value="1" enum="ScreenSpaceAA">
-			Use fast approximate antialiasing. FXAA is a popular screen-space antialiasing method, which is fast but will make the image look blurry, especially at lower resolutions. It can still work relatively well at large resolutions such as 1440p and 4K.
+			Use fast approximate antialiasing. FXAA is a popular screen-space antialiasing method, which is fast but will make Sthe image look blurry, especially at lower resolutions. It can still work relatively well at large resolutions such as 1440p and 4K.
 		</constant>
 		<constant name="SCREEN_SPACE_AA_MAX" value="2" enum="ScreenSpaceAA">
 			Represents the size of the [enum ScreenSpaceAA] enum.
 		</constant>
+		<constant name="FSR_UPSCALE_DISABLED" value="0" enum="FSRUpscaleQuality">
```

Comment:
I know that FSR is promoted as having these 4 Modes, However using a float that determines how much lower res you're going with makes more sense instead of hard coding AMD's Predefined quality settings.

These Quality settings Should be noted in the Description, as a reference, but it's up to the devs to determine if they wish to follow AMD's Suggestion, or let the user pick the resolution scaling (like DOTA 2 does)

Reply:
In its integration guidelines, AMD recommends exposing only those 4 presets. If you really want to expose custom settings, you could add a `FSR_UPSCALE_CUSTOM` value which would then read the value of a float project setting to use as a resolution scale.

Reply:
I think using the 4 presets also aligns with the ethos of Godot: usability and ease of use first. Users won't have to check out the docs to see what the values mean if we use the presets. Also, it helps with standardization with other game engines.

Reply:
Added `FSR_UPSCALE_CUSTOM` and `fsr_upscale_custom_quality` to add support for custom qualities